mrp_subcontracting_partner_management/models/res_partner.py

```diff
@@ -6,258 +8,234 @@ class ResPartner(models.Model):
+        - Location;
+        - Operation type;
+        - Route Rule for Subcontracting buy;
+        - Route Rule for Subcontracting resupply.
         """
-        for record in self:
-            record._set_subcontracting_values_active(False)
-        result = super(ResPartner, self).unlink()
-        return result
+        partners = self.filtered(lambda p: p.is_subcontractor_partner)
+        field_names = [*self.get_data_struct(), "property_stock_subcontractor"]
+        field_names.remove("partner_picking_type_id")
+        for field in field_names:
+            records = partners.mapped(field)
+            records.write({"name": name})
+            self._update_name_translation(records, name)
+        type_name = "%s:  IN" % name
+        code = "".join(re.findall(r"\b\w", type_name))
+        picks = partners.mapped("partner_picking_type_id")
+        picks.write({"name": type_name, "sequence_code": code})
+        self._update_name_translation(picks, type_name)
 
-    def write(self, values):
-        for record in self:
-            is_subcontractor_partner = values.get("is_subcontractor_partner")
-            active = values.get("active")
-            if is_subcontractor_partner is not None:
-                values.update(record._update_subcontractor_entities_for_record(values))
-            if active is not None:
-                record._set_subcontracting_values_active(active)
-        super(ResPartner, self).write(values)
+    def unlink(self):
+        """This Method is override to archive all subcontracting field"""
+        self._set_subcontracting_values_active(False)
+        return super(ResPartner, self).unlink()
+
+    def write(self, vals):
+        if "is_subcontractor_partner" in vals:
+            self._update_subcontractor_entities_for_record(
+                vals.get("is_subcontractor_partner")
+            )
+        if "active" in vals:
+            self._set_subcontracting_values_active(vals.get("active"))
+        result = super(ResPartner, self).write(vals)
+        if vals.get("name"):
+            self._update_subcontractor_values_name(vals.get("name"))
+        return result
 
-    @api.model
-    def create(self, values):
-        partner = super(ResPartner, self).create(values)
-        if values.get("is_subcontractor_partner", False):
-            partner._create_subcontractor_entities()
-        return partner
+    @api.model_create_multi
+    def create(self, vals_list):
+        records = super(ResPartner, self).create(vals_list)
+        check_data = self.get_data_struct().items()
+        for record in records.filtered(
+            lambda r: r.is_subcontractor_partner and r.is_company
+        ):
+            values = {}
+            for key, func in check_data:
+                if not getattr(record, key) or not values.get(key):
+                    values.update(**getattr(record, func)(values) or {})
+            if values:
+                record.write(values)
```

Comment:
This can be done before calling super iterating on each dictionary and seeing if `is_subcontractor_partner` and `is_company` vals are set.